internal/pkg/otel/manager/execution_subprocess.go

```diff
@@ -74,7 +74,20 @@ type subprocessExecution struct {
 
 // startCollector starts a supervised collector and monitors its health. Process exit errors are sent to the
 // processErrCh channel. Other run errors, such as not able to connect to the health endpoint, are sent to the runErrCh channel.
-func (r *subprocessExecution) startCollector(ctx context.Context, baseLogger *logger.Logger, logger *logger.Logger, cfg *confmap.Conf, processErrCh chan error, statusCh chan *status.AggregateStatus, forceFetchStatusCh chan struct{}) (collectorHandle, error) {
+func (r *subprocessExecution) startCollector(
+	ctx context.Context,
+	logLevel string,
+	baseLogger *logger.Logger,
+	logger *logger.Logger,
+	cfg *confmap.Conf,
+	processErrCh chan error,
+	statusCh chan *status.AggregateStatus,
+	forceFetchStatusCh chan struct{},
+) (collectorHandle, error) {
+	if err := r.setLogLevelFromString(logLevel); err != nil {
```

Comment:
It seems like the log level set with `newSubprocessExecution` is pointless as it just gets immediately overwritten.

I would remove it the argument from `newSubprocessExecution` and also from the `subprocessExecution` struct. This will prevent someone from accidentally accessing the copy in the struct and assuming it is always equal to what the collector is going to execute with. 

There is a window right now between init and the collector starting where you can access the log level in a method on `subprocessExecution` but the level you read is not guaranteed to be the one the collector starts with.